core/variant/variant_call.cpp

```diff
@@ -1366,30 +1366,43 @@ static void _register_variant_builtin_methods() {
 	bind_method(String, naturalnocasecmp_to, sarray("to"), varray());
 	bind_method(String, length, sarray(), varray());
 	bind_method(String, substr, sarray("from", "len"), varray(-1));
-	bind_methodv(String, find, static_cast<int (String::*)(const String &, int) const>(&String::find), sarray("what", "from"), varray(0));
-	bind_method(String, count, sarray("what", "from", "to"), varray(0, 0));
-	bind_method(String, countn, sarray("what", "from", "to"), varray(0, 0));
-	bind_method(String, findn, sarray("what", "from"), varray(0));
-	bind_method(String, rfind, sarray("what", "from"), varray(-1));
-	bind_method(String, rfindn, sarray("what", "from"), varray(-1));
-	bind_method(String, match, sarray("expr"), varray());
-	bind_method(String, matchn, sarray("expr"), varray());
+	bind_methodv(String, find, static_cast<int (String::*)(const String &, int) const>(&String::find), sarray("substring", "from"), varray(0));
+	bind_methodv(String, findn, static_cast<int (String::*)(const String &, int) const>(&String::findn), sarray("substring", "from"), varray(0));
+	bind_methodv(String, rfind, static_cast<int (String::*)(const String &, int) const>(&String::rfind), sarray("substring", "from"), varray(-1));
+	bind_methodv(String, rfindn, static_cast<int (String::*)(const String &, int) const>(&String::rfindn), sarray("substring", "from"), varray(-1));
```

Comment:
What's the functional difference between the previous binding and the new ones with `bind_methodv`, aside from the ugly casting? :)

Reply:
This beautiful casting that you see right here (:p) tells the compiler which overloaded function to use.

The underlying cause is that in this PR I've overloaded all these functions with equivalent implementations that make use of C-strings (`const char *`) for performance reasons. On a more concrete example, just consider the following two function signatures:

```cpp
String String::find(const String &)
String String::find(const char *)
```

From a GDScript standpoint, nothing changes since the casting above chooses the right function signature which is the first one you see above, but from the C++ side, it makes the engine more efficient by avoiding `String` object allocations (which in turn avoid `CowData<T>` object allocations, plus `malloc` that in turn fire system calls to allocate the memory the `String` is going to use to the `char32_t` string itself). So from the C++ side, calls to `String::find("my_pattern")` are more effective, but the same function can also be called with `String` objects instead of `const char *` obviously.